src/metabase/models/params/chain_filter.clj

```diff
@@ -517,6 +518,14 @@
     field-id
     (field-values/field-should-have-field-values? field-id)))
 
+(defn- check-field-value-query-permissions!
+  "Check query permissions against the chain-filter-mbql-query (private #196)"
+  [field-id constraints options]
+  (->> (chain-filter-mbql-query field-id constraints options)
+         qp/preprocess
+         qp.perms/check-query-permissions*)
+  true)
```

Comment:
I don't think this really deserves an `!` since it's not done for the sort of side effects that [aren't safe in an STM context](https://guide.clojure.style/#naming-unsafe-functions), i.e. if you call this function 10 times it doesn't make any difference vs calling it once. It's still a pure function even if we're only calling it so maybe it will throw an exception.

I think calling it `check-field-value-query-permissions` is fine, that's generally the pattern we follow for stuff like this e.g. `check-valid-search-field` in this same namespace